app/Console/Commands/PrintIcalYearlyClosingPeriod.php

```diff
@@ -0,0 +1,165 @@
+            $this->output->writeln("UID:{$this->uuid()}");
+            $this->output->writeln("DTSTART;VALUE=DATE:{$from->format(self::DATE_FORMAT_ICAL)}");
+            $this->output->writeln("DTEND;VALUE=DATE:{$to->format(self::DATE_FORMAT_ICAL)}");
+            $this->output->writeln("SUMMARY;LANGUAGE=nl-BE:{$name}");
+            $this->output->writeln("END:VEVENT");
+            $from = $from->add(new DateInterval("P1Y"));
+            $to = $to->add(new DateInterval("P1Y"));
+        }
+    }
+
+    /**
+     * Parse the given command line arguments {from} and {to}
+     *
+     * @param $from string The from date
+     * @param $to string The to date
+     *
+     * @return array<string, DateTimeImmutable, DateTimeImmutable>
+     *  Result of the arg parse, first element is an error message.
+     *  If the error is `null`, there were no parse errors.
+     *  Next two elements are the from date and to date.
+     *  These will be `null` if there is an error.
+     */
+    protected function parseArgs(string $from, string $to) : array
+    {
+        $errAllowedFormats = "(YYYY-MM-DD or YYYYMMDD)";
+        $dashCount = substr_count($from, "-");
+        if ($dashCount != substr_count($to, "-")) {
+            return ["From- and To date should be in the same format $errAllowedFormats", null, null];
+        }
+        if ($dashCount != 2 && $dashCount !=0) {
+            return ["Only $errAllowedFormats are considered valid date formats", null, null];
+        }
+        $expectedLen = $dashCount == 2 ? 10 : 8;
+        if (strlen($from) != $expectedLen) {
+            $longShort = strlen($from) < $expectedLen ? "short" : "long";
+            return ["Only $errAllowedFormats format allowed, the from date is too " . $longShort, null, null];
+        }
+        if (strlen($to) != $expectedLen) {
+            $longShort = strlen($to) < $expectedLen ? "short" : "long";
+            return ["Only $errAllowedFormats format allowed, the to date is too " . $longShort, null, null];
+        }
+        $dateFrom = null;
+        $dateTo = null;
+        $dti = new \DateTimeImmutable();
+        $format = $dashCount == 2 ? "Y-m-d" : "Ymd";
+        $dateFrom = $dti->createFromFormat($format, $from);
+        if (!$dateFrom) {
+            return ["Invalid from date", null, null];
+        }
+        $dateTo = $dti->createFromFormat($format, $to);
+        if (!$dateTo) {
+            return ["Invalid to date", null, null];
+        }
+        if ($dateFrom >= $dateTo) {
+            return ["The from date must be earlier than the to date", null, null];
+        }
+        if ($dateFrom->format("Y") >= self::MAX_YEAR || $dateTo->format("Y") >= self::MAX_YEAR) {
+            return ["This command does not handle dates beyond " . self::MAX_YEAR, null, null];
+        }
+        return [null, $dateFrom, $dateTo];
+    }
+
+    /**
+     * Generate a (pseudo) UUID
+     * This might not be a _real_ UUID, but for all intended purposes, it'll be
```

Comment:
While fancy and functional, this project already has dependencies on two components that can generate uuids:
https://symfony.com/doc/current/components/uid.html
and
https://github.com/ramsey/uuid
So we might as well use one of those here

Reply:
Fixed in [efec801b81f015fecbb6cdb1329cfaf900d46833]